base/point.h

```diff
@@ -25,13 +25,19 @@ class Point {
    * \brief Standard destructor.
    */
   virtual ~Point();
-  /** General accessor method for the underlying data.
+
+  /** General accessor method for each feature of the underlying data.
    * \param feature_id The coordinate of interest. Indexing is implementation
    * specific.
    * \return The value of the coordinate specified in the input.
    */
   virtual double GetFeatureValue(unsigned feature_id) const = 0;
 
+  /** General accessor method for the label of the underlying data.
+   * \return The label of the object.
+   */
+  virtual double GetLabel() const = 0;
```

Comment:
Its the class/relevance of each point. If the object has no label, we can simply either ignore the field or insert some dummy value, such as -1.